### ISSUE TITLE
Develop cost function and get data into a scoreable form

### ISSUE THREAD

**Issue by k-j-m** [k-j-m/CourseAnalysis#3]
We've pulled all of the data together and built all of the indexes we need to relate one dataset to another, now we need to start doing something useful with it....

**Comment by k-j-m** [comment]
Closed. Cost function (with gradients) working successfully. Annoyingly had to write my own gradient descent solver which works and isn't too complex, but is VERY sensitive to learning rate.
